CONTRIBUTING.md

```diff
@@ -83,6 +83,10 @@ In addition to the conventions covered in the SPIFFE project's
 [CONTRIBUTING](https://github.com/spiffe/spiffe/blob/master/CONTRIBUTING.md), the following
 conventions apply to the SPIRE repository:
 
+## SQL Plugin Changes
+
+Datastore changes must be present in at least one full major release cycle prior to introducing code changes that depend on them.
```

Comment:
currently does not match actual logic (checking minor versions)
waiting for @evan2645 for which direction the logic should be pre-1.0